localstack/services/apigateway/templates.py

```diff
@@ -283,9 +282,16 @@ def render(self, api_context: ApiInvocationContext, **kwargs) -> Union[bytes, st
 
         # we render the template with the context data and the response content
         variables = self.build_variables_mapping(api_context)
-        rendered_tpl = self.render_vtl(template, variables=variables)
+        response._content = self._render_json_result(template, variables)
 
-        # TODO: we need to do a parity test for templates that generate invalid JSON
-        response._content = json.dumps(ast.literal_eval(rendered_tpl.strip()))
-        LOG.info("Endpoint response body after transformations:\n%s", response._content)
+        LOG.debug("Endpoint response body after transformations:\n%s", response._content)
         return response._content
+
+    def _render_json_result(self, template: str, variables: dict[str, Any]) -> str:
```

Comment:
nit: we only support json templates atm, so this helper method is indeed a good idea I'd just consider a name like `_render_as_json` so we could apply this to others like `_render_as_text`

Reply:
Good point, updated 👍